plugin/src/test/java/hudson/plugins/swarm/SwarmClientIntegrationTest.java

```diff
@@ -471,6 +474,30 @@ public void jarCacheWithCustomPath() throws Exception {
                 jarCachePath);
     }
 
+    @Test
+    public void metricsPrometheus() throws Exception {
+        swarmClientRule.createSwarmClient("-prometheusPort", "9999");
+
+        // Fetch the metrics page from the client
+        URL url = new URL("http://localhost:9999/prometheus");
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("GET");
+        BufferedReader reader =
+                new BufferedReader(new InputStreamReader(connection.getInputStream()));
+        String inputLine;
+        StringBuilder content = new StringBuilder();
+        while ((inputLine = reader.readLine()) != null) {
+            content.append(inputLine);
+        }
+        reader.close();
+        connection.disconnect();
```

Comment:
In Java 7 and later, you can avoid the need to explicitly `close()` these resources by using `URL#openStream` and try-with-resources:

```suggestion
        // Fetch the metrics page from the client
        StringBuilder content = new StringBuilder();
        try (InputStream is = new URL("http://localhost:9999/prometheus").openStream();
                BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
            String inputLine;
            while ((inputLine = reader.readLine()) != null) {
                content.append(inputLine);
            }
        }
```

Reply:
Thanks for the suggestion; my Java is a bit rusty. :sweat_smile: